pctasks/core/pctasks/core/models/item.py

```diff
@@ -0,0 +1,95 @@
+from datetime import datetime
+from typing import Any, Dict, Optional
+
+import pystac
+from pydantic import Field, validator
+
+from pctasks.core.models.record import Record
+from pctasks.core.utils import StrEnum
+
+
+class ItemRecordType(StrEnum):
+    STAC_ITEM = "StacItem"
+    ITEM_UPDATED = "ItemUpdated"
+
+
+class ItemRecord(Record):
+    type: ItemRecordType
+    stac_id: str
+
+    @validator("stac_id")
+    def _stac_id_validator(cls, v: str) -> str:
+        # Ensure a single forward slash in the string
+        if v.count("/") != 1:
+            raise ValueError("stac_id must contain a single forward slash")
+        return v
+
+    def get_id(self) -> str:
+        version = getattr(self, "version", "")
```

Comment:
Ugly, but the best way I could think of to convince mypy that this would be OK (normally, I would use a mixin with a typed class attribute, but that conflicts with how pydantic does stuff).